File: extensions/shared/src/main/java/app/morphe/extension/shared/spoof/potoken/PoTokenException.java
/*
 * Copyright 2026 Morphe.
 * https://github.com/MorpheApp/morphe-patches/pull/2533
 *
 * See the included NOTICE file for GPLv3 Section 7 terms that apply to this code.
 */

package app.morphe.extension.shared.spoof.potoken;

public class PoTokenException extends Exception {
    public PoTokenException(String message) {
        super(message);
    }

    public static class BadWebViewException extends Exception {
        public BadWebViewException(String message) {
            super(message);
        }
    }

    public static Exception buildExceptionForJsError(String error) {
        if (error != null && error.contains("SyntaxError")) {
            return new BadWebViewException(error);
        }
        return new PoTokenException(error);
    }
}

File: extensions/shared/src/main/java/com/liskovsoft/youtubeapi/app/potokennp2/PoTokenException.kt
package com.liskovsoft.youtubeapi.app.potokennp2

internal class PoTokenException(message: String) : RuntimeException(message)

// to be thrown if the WebView provided by the system is broken
internal class BadWebViewException(message: String) : RuntimeException(message)

internal fun buildExceptionForJsError(error: String): Throwable {
    return if (error.contains("SyntaxError"))
        BadWebViewException(error)
    else
        PoTokenException(error)
}
